.github/workflows/ci.yml

```diff
@@ -21,17 +21,26 @@ jobs:
       - uses: actions/setup-python@v5
       - name: Install dependencies
         run: |
+          python -m venv venv
+          source venv/bin/activate
+          python -m pip install --upgrade pip
           python -m pip install pre-commit
           python -m pip freeze --local
           lsblk -l
           echo "sda rotational = $(cat /sys/block/sda/queue/rotational)"
           echo "sdb rotational = $(cat /sys/block/sdb/queue/rotational)"
+      - name: Install dependencies
+        run: |
+          source venv/bin/activate
+          pip install pre-commit
```

Comment:
why again?

Reply:
because among different `run` commands you need to `source` again

Reply:
(I am testing this not 100% sure about this)